modules/internal/src/main/scala/encoders/avro.scala

```diff
@@ -63,7 +68,13 @@ object avro extends AvroMarshallers {
 
     }
 
-  object bigdecimal {
+  @deprecated(
+    "This avro4s encoder doesn't fulfill the avro specs. " +
+      "Be aware that this serializes without taking into account the precision and scale, " +
+      "and it could lead to serialization problems if the client uses a different serializer.",
+    "0.15.1"
+  )
+  object bigDecimal {
```

Comment:
you're breaking bincompat here by making the `D` in `Decimal` uppercase. I think it's a typo :)